dashboard-api/internal/types/studyconfig_test.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestStudyConfigType(t *testing.T) {
+	assert := require.New(t)
+	t.Run("can transofrm an entire study config for database", func(t *testing.T) {
+		input := StudyConfig{
+			StudyID: "foobar",
+			UserID:  "baz",
+			General: GeneralConfig{
+				Name:             "Foo",
+				AdAccount:        "12345",
+				DestinationType:  "Web",
+				OptInWindow:      48,
+				OptimizationGoal: "link_clicks",
+				PageID:           "1",
+				MinBudget:        1,
+			},
+			Targeting: TargetingConfig{
+				TemplateCampaignName: "Bar",
+				DistributionVars:     Location,
+			},
+			TargetingDistribution: TargetingDistributionConfig{
+				Age:      "21",
+				Gender:   "F",
+				Location: "Spain",
+			},
+		}
+		expected := []DatabaseStudyConf{
+			{
+				StudyID:  "foobar",
+				ConfType: "general",
+				Conf:     []byte(`{"name":"Foo","objective":"","optimization_goal":"link_clicks","destination_type":"Web","page_id":"1","min_budget":1,"opt_window":48,"instagram_id":"","ad_account":"12345"}`),
+			},
+			{
+				StudyID:  "foobar",
+				ConfType: "targeting",
+				Conf:     []byte(`{"template_campaign_name":"Bar","distribution_vars":"location"}`),
+			},
+			{
+				StudyID:  "foobar",
+				ConfType: "targeting_distribution",
+				Conf:     []byte(`{"age":"21","gender":"F","location":"Spain"}`),
+			},
+		}
+		s, err := input.TransformForDatabase()
+		assert.NoError(err)
+		for i, _ := range expected {
+			assert.Equal(expected[i], s[i])
+		}
+	})
+	t.Run("does not transform null configs", func(t *testing.T) {
```

Comment:
Cool! Only thing I don't grasp is how the "null" test is different to the happy path except for `assert.Equal(len(s), 2)` can you show me what that means?

Reply:
Yes lets go through it tomorrow morning?